v4l2/v4l2_decode.cpp

```diff
@@ -39,12 +39,12 @@
 #include "egl/egl_vaapi_image.h"
 
 #define INT64_TO_TIMEVAL(i64, time_val) do {            \
-        time_val.tv_sec = (int32_t)(i64 >> 32);         \
-        time_val.tv_usec = (int32_t)(i64 & 0xffffffff); \
+        time_val.tv_sec = (int32_t)(i64 >> 31);         \
+        time_val.tv_usec = (int32_t)(i64 & 0x7fffffff); \
     } while (0)
 #define TIMEVAL_TO_INT64(i64, time_val) do {            \
         i64 = time_val.tv_sec;                          \
-        i64 = (i64 << 32) + time_val.tv_usec;           \
+        i64 = (i64 << 31)  + time_val.tv_usec;  \
```

Comment:
31 is magic number, people who do not blame the code will guess, why not use 32 etc.
Since manual said time_val.tv_usec never larger than 1000000, Do you think it's ok to use  `i64 = time_val.tv_sec*1000000 + time_val.tv_usec`, and `time_val.tv_sec = i64/1000000; time_val.tv_usec = i64%1000000`